src/Magento/FunctionalTestingFramework/Console/GenerateTestsCommand.php

```diff
@@ -65,14 +65,18 @@ protected function execute(InputInterface $input, OutputInterface $output)
     {
         $tests = $input->getArgument('name');
         $config = $input->getOption('config');
-        $json = $input->getOption('tests');
+        $json = $input->getOption('tests'); // for backward compatibility
         $force = $input->getOption('force');
         $time = $input->getOption('time') * 60 * 1000; // convert from minutes to milliseconds
         $debug = $input->getOption('debug') ?? MftfApplicationConfig::LEVEL_DEVELOPER; // for backward compatibility
         $remove = $input->getOption('remove');
         $verbose = $output->isVerbose();
         $allowSkipped = $input->getOption('allowSkipped');
 
+        if (!empty($tests)) {
+            $json = $this->getTestAndSuiteConfiguration($tests);
```

Comment:
There's quite a bit of weird little implications of making this change (I don't disagree with the change though).

Due to the way TestGenerator uses a local `$this-tests` variable for some reason, and due to the fact that we are now *always* passing in something to that variable, I would suggest the following:

* Pass in the *entire* testsConfiguration in `ln 96`
* Run through TestGenerator's uses of `$this->tests` and refactor use so that it understands the new config being passed in
* Unit test the following usecases
** Test0 (no suite) Test1 (suite1)
** generate:tests (everything is generated)
** generate:tests Test0 (only Test0 is generated)
** generate:tests Test1 (only Test1 is generated)

I started off seeing incorrect behavior with testcase 3 which led me down a rabbit hole of playing whackamole with changes, and I think you will just need to deep dive into TestGenerator and do refactoring of how it uses the `$this->tests` variable.

Reply:
Put in a fix by initializing the json configuration for `tests` and `suites` to null instead of [] which generates/runs all tests/suites in `getTestAndSuiteConfiguration(array $tests)`